ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -3017,12 +3017,12 @@ impl CompiledTrace for X64CompiledTrace {
         };
         // Check that the target address is 8-byte aligned. This is required so we can patch
         // this atomically.
-        let patch_addr = patch_addr as *mut u64;
-        debug_assert!(patch_addr.addr() % 8 == 0);
+        debug_assert!(patch_addr % 8 == 0);
+        let patch_addr = std::sync::atomic::AtomicPtr::new(patch_addr as *mut u64);
         if res == 0 {
             // Patch the new 64-bit address into the mov instruction.
-            let addr_in_bytes = staddr as u64;
-            unsafe { *patch_addr = addr_in_bytes };
+            let val = &mut (staddr as u64);
```

Comment:
Does `&mut` actually work here? I would have expected `unsafe` to be needed.

I'm also not entirely sure this does what we think it should. I *think* (but I might be wrong) that what is this doing is the equivalent of:

```c
uint_ptr_t *patch_addr = ...;
patch_addr = val;
```

i.e. I *think* you're expecting this to be `*patch_addr = val` but I don't think it's doing that additional level of indirection.

Reply:
The `AtomicPtr::store` instruction expects a pointer to the new value. That's why I had to turn `staddr` into a pointer. I wouldn't expect the code to run if this doesn't write the `staddr` into the `patch_addr` properly.

Reply:
Don't flatter my ego :p